app/lib/cfe_connection.rb

```diff
@@ -124,13 +124,16 @@ def api_result(assessment_id)
 private
 
   def create_record(assessment_id, record_type, record_data)
-    cfe_connection.post "/assessments/#{assessment_id}/#{record_type}", record_data
+    url = "/assessments/#{assessment_id}/#{record_type}"
+    response = cfe_connection.post url, record_data
+    return if response.status == 200
+
+    raise "Call to CFE url #{url} returned status #{response.status} and message:\n#{response.body}"
```

Comment:
Sorry I just realised that api_result can fail too (it fails with a 422) when something goes wrong.
Also IMHO it might be better to check result for success (result.success? I think) as POST requests are often *meant* to return 201 rather than 200 (although I think CFE doesn't do this currently)